Makefile

```diff
@@ -1,40 +1,46 @@
-.PHONY: install clean package ltvalues
-
-DIST_ROOT=dist
-KUBE_INSTALL := $(shell command -v kubectl 2> /dev/null)
-HELM_INSTALL := $(shell command -v helm 2> /dev/null)
-
-SUB_CHARTS := $(shell ls mattermost-helm/charts)
-
-all: package
-
-check:
-ifndef KUBE_INSTALL
-    $(error "kubectl is not available please install from https://kubernetes.io/")
-endif
-ifndef HELM_INSTALL
-    $(error "helm is not available please install from https://github.com/kubernetes/helm")
-endif
-
-.init:
-	helm init --client-only
-	touch $@
-
-package: check .init
-	# Temporary while we wait for changes to be merged upstream
-	helm repo add mattermost-mysqlha https://releases.mattermost.com/helm/mysqlha
-	mkdir -p dist
-	rm -f mattermost-helm/charts/*.tgz
-	helm package mattermost-helm/charts/* -d mattermost-helm/charts
-	helm dep up mattermost-helm
-	helm package mattermost-helm -d $(DIST_ROOT)
-
-clean:
-	rm -rf $(DIST_ROOT)
-	rm -f mattermost-helm/charts/*.tgz
-
-install: check
-	helm install dist/mattermost-helm*.*
-
-ltvalues:
-	cp ../platform-private/kubernetes/values_loadtest.yaml mattermost-helm/values.yaml
+DOCKER := $(shell which docker)
+DOCKER_OPTS += --rm -u $$(id -u):$$(id -g)
+CLUSTER_NAME ?= mattermost-helm-test
+
+DOCKER_IMAGE_SHELLCHECK ?= koalaman/shellcheck:v0.8.0@sha256:4c4427336d2b4bdb054a1e97396fa2e9ca0c329a1f43f831b99bcaae4ac24fcd
+# Official image of chart-testing[https://github.com/helm/chart-testing].
+# Contains chart-testing cli,  helm and kubectl.
+DOCKER_IMAGE_CT ?= quay.io/helmpack/chart-testing:v3.7.0@sha256:2f87e56a0cebc6d9bb78d51cb6adac97858afb0887cac4d65f7f8c6a16054568
+
+VERIFY_SCRIPTS += /src/tests/e2e-kind.sh
+
+CT_CONFIG ?= tests/ct.yaml
+KIND_CONFIG ?= tests/kind-config.yaml
+CLUSTER_NAME ?= mattermost-helm-test
+
```

Comment:
A proposal for maybe decoupling these to more generic targets. 
Like `make test` `make lint` . 

To have standardized targets for Makefiles . It will be easy to also add additional tests/lints in the future.